tests/hook/enumerate.c

```diff
@@ -0,0 +1,83 @@
+#include "clar_libgit2.h"
+
+static git_repository *g_repo = NULL;
+
+void test_hook_enumerate__initialize(void)
+{
+	g_repo = cl_git_sandbox_init("testrepo");
+}
+
+void test_hook_enumerate__cleanup(void)
+{
+	cl_git_sandbox_cleanup();
+	g_repo = NULL;
+}
+
+static int test_git_hook_foreach_cb(const char *hook_name, void *payload)
+{
+	git_vector *hook_list = payload;
+	git_vector_insert(hook_list, (char *)hook_name);
+	return 0;
+}
+
+void test_hook_enumerate__foreach_hooks(void)
+{
+	git_vector hook_list = GIT_VECTOR_INIT;
+	git_vector expected_list = GIT_VECTOR_INIT;
+
+	git_vector_insert(&expected_list, "commit-msg");
+	git_vector_insert(&expected_list, "post-merge");
+
+	cl_git_pass(git_hook_foreach(g_repo, test_git_hook_foreach_cb, &hook_list));
+
+	cl_assert_equal_v_str(&hook_list, &expected_list);
+
+	git_vector_free(&hook_list);
+	git_vector_free(&expected_list);
+}
+
+#define ALT_HOOK_DIR "../testhooks"
+
+void test_hook_enumerate__foreach_hooks_config_override(void)
+{
+	git_config *cfg;
+	git_buf alt_hook = GIT_BUF_INIT;
+	char *alt_hook_str;
+	git_vector hook_list = GIT_VECTOR_INIT;
+	git_vector expected_list = GIT_VECTOR_INIT;
+
+	/* Change the repository hooks  */
+	cl_git_pass(git_repository_config(&cfg, g_repo));
+	cl_git_pass(git_config_set_string(cfg, "core.hooksPath", ALT_HOOK_DIR));
+
+	cl_git_pass(git_hook_dir(&alt_hook_str, g_repo));
+	cl_git_pass(git_buf_sets(&alt_hook, alt_hook_str));
```

Comment:
You're leaking `alt_hook_str`. You probably want `git_buf_attach(&alt_hook, alt_hook_str, strlen(alt_hook_str) + 1)` here, which will also free the string later on.